jetty-webapp/src/test/java/org/eclipse/jetty/webapp/TempDirTest.java

```diff
@@ -0,0 +1,265 @@
+        server.start();
+        tempDirectory = webapp.getTempDirectory();
+        assertNotNull(tempDirectory);
+        assertTrue(tempDirectory.exists());
+        assertTrue(tempDirectory.isDirectory());
+
+        // Once server is stopped the WebApp temp should be deleted if persistTempDir is false.
+        server.stop();
+        tempDirectory = webapp.getTempDirectory();
+        assertThat(tempDirectory != null && tempDirectory.exists(), is(persistTempDir));
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testPreDefinedTempDirectory(boolean persistTempDir) throws Exception
+    {
+        setupServer();
+        webapp.setPersistTempDirectory(persistTempDir);
+
+        // The temp directory is defined but has not been created.
+        File webappTempDir = MavenTestingUtils.getTargetTestingPath("webappTempDir").toFile();
+        IO.delete(webappTempDir);
+        webapp.setTempDirectory(webappTempDir);
+        assertThat(webapp.getTempDirectory(), is(webappTempDir));
+        assertFalse(webappTempDir.exists());
+
+        // Once server is started the WebApp temp directory exists and is valid directory.
+        server.start();
+        File tempDirectory = webapp.getTempDirectory();
+        assertNotNull(tempDirectory);
+        assertTrue(tempDirectory.exists());
+        assertTrue(tempDirectory.isDirectory());
+
+        // Once server is stopped the WebApp temp should be deleted if persistTempDir is false.
+        server.stop();
+        tempDirectory = webapp.getTempDirectory();
+        assertThat(tempDirectory != null && tempDirectory.exists(), is(persistTempDir));
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testPreExistingTempDirectory(boolean persistTempDir) throws Exception
+    {
+        setupServer();
+        webapp.setPersistTempDirectory(persistTempDir);
+
+        // The temp directory is defined and has already been created.
+        File webappTempDir = MavenTestingUtils.getTargetTestingPath("webappTempDir").toFile();
+        IO.delete(webappTempDir);
+        if (!webappTempDir.exists())
+            assertTrue(webappTempDir.mkdir());
+        webapp.setTempDirectory(webappTempDir);
+        assertThat(webapp.getTempDirectory(), is(webappTempDir));
+        assertTrue(webappTempDir.exists());
+
+        // Once server is started the WebApp temp directory exists and is valid directory.
+        server.start();
+        File tempDirectory = webapp.getTempDirectory();
+        assertNotNull(tempDirectory);
+        assertTrue(tempDirectory.exists());
+        assertTrue(tempDirectory.isDirectory());
+
+        // Once server is stopped the WebApp temp should be deleted if persistTempDir is false.
+        server.stop();
+        tempDirectory = webapp.getTempDirectory();
+        assertThat(tempDirectory != null && tempDirectory.exists(), is(persistTempDir));
+    }
```

Comment:
Need tests for using the WebAppContext.BASETEMPDIR, and for using $jetty.base/work dir.

Reply:
There are now tests for both `WebAppContext.BASETEMPDIR` and `$jetty.base/work`.